src/Radio.js

```diff
@@ -0,0 +1,70 @@
+import classNames from 'classnames';
+import React from 'react';
+import warning from 'warning';
+
+import bootstrapUtils, { bsClass } from './utils/bootstrapUtils';
+
+const propTypes = {
+  inline: React.PropTypes.bool,
+  disabled: React.PropTypes.bool,
+  /**
+   * Only valid if `inline` is not set.
+   */
+  validationState: React.PropTypes.oneOf(['success', 'warning', 'error']),
+};
+
+const defaultProps = {
+  inline: false,
+  disabled: false,
+};
+
+class Radio extends React.Component {
+  render() {
+    const {
+      inline, disabled, validationState, className, style, children, ...props,
+    } = this.props;
+
+    delete props.bsClass;
+
+    if (inline) {
+      const classes = {
+        [bootstrapUtils.prefix(this.props, 'inline')]: true,
+        disabled,
+      };
+
+      warning(
+        !validationState,
+        '`validationState` is invalid on `<Radio inline>`.'
```

Comment:
Maybe the warning should suggest wrapping inline checkboxes in a FormGrop with validation state instead?